subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/JavaCompilerIntegrationSpec.groovy

```diff
@@ -58,6 +58,26 @@ abstract class JavaCompilerIntegrationSpec extends BasicJavaCompilerIntegrationS
         file("build/classes/main/compile/test/Person.class").exists()
     }
 
+    def compileWithCustomDiagnosticListener() {
```

Comment:
Could you also add a test case that verifies that the listener is not called if the compiler process is forked?